conjure-java-core/src/integrationInput/java/com/palantir/product/EnumExample.java

```diff
@@ -20,7 +21,7 @@
  * compile time.
  */
 @Generated("com.palantir.conjure.java.types.EnumGenerator")
-public final class EnumExample {
+public final class EnumExample implements ConjureEnum<EnumExample> {
```

Comment:
why is it necessary for this to satisfy the `ConjureEnum` interface?

Reply:
This allows for the creation of generic utility code as given in the CR description. Right now, I have to write ugly, cludgy, or even fragile code to work with conjure enums generically.